Update FDS README (#2605)
adap/flower: datasets/README.md

# Usage

> The Flower Datasets exposes `FederatedDataset` abstraction to represent the dataset needed for federated learning/analytics. It has two powerful methods that let you handle the dataset preprocessing. They are `load_partition(node_id, split)` and `load_full(split)`.

Reviewer: ```suggestion
Flower Datasets exposes the `FederatedDataset` abstraction to represent the dataset needed for federated learning/evaluation/analytics. It has two powerful methods that let you handle the dataset preprocessing: `load_partition(node_id, split)` and `load_full(split)`.
```

> Here's the most basic quick example of how to partition the MNIST dataset:

Reviewer: ```suggestion
Here's a basic quickstart example of how to partition the MNIST dataset:
```

centralized_data = mnist_fds.load_full("test")
```

For more details, please refer to the specific how-to guides or tutorial. They showcase customization and more advanced features.

# Future release

Here are a few of the things that we will work on in future releases:

> * ✅Support for more datasets (especially the ones that have user id present).
> * ✅Creation of custom `Partitioner`s.
> * ✅More out-of-the-box `Partitioner`s.
> * ✅Passing `Partitioner`s via `FederatedDataset`'s `partitioner` argument. 
> * ✅Customization of the dataset splitting before the partitioning.

Reviewer: ```suggestion
* ✅ Support for more datasets (especially the ones that have user id present).
* ✅ Creation of custom `Partitioner`s.
* ✅ More out-of-the-box `Partitioner`s.
* ✅ Passing `Partitioner`s via `FederatedDataset`'s `partitioners` argument. 
* ✅ Customization of the dataset splitting before the partitioning.
```
